src/validate/validate_state.test.ts

```diff
@@ -0,0 +1,149 @@
+import {validateState} from './validate_state';
+
+describe('Validate state', () => {
+    test('Should return error if type is not an object', () => {
+        let errors = validateState({key: 'state', value: 3});
+        expect(errors).toHaveLength(1);
+        expect(errors[0].message).toBe('state: object expected, number found');
+
+        errors = validateState({key: 'state', value: true});
+        expect(errors).toHaveLength(1);
+        expect(errors[0].message).toBe('state: object expected, boolean found');
+
+        errors = validateState({key: 'state', value: null});
+        expect(errors).toHaveLength(1);
+        expect(errors[0].message).toBe('state: object expected, null found');
+    });
+
+    test('Should pass if type is object', () => {
+        const errors = validateState({key: 'state', value: {}});
+        expect(errors).toHaveLength(0);
+    });
+
+    describe('number', () => {
+        test('should return no errors for correct definition', () => {
+            const errors = validateState({
+                key: 'state',
+                value: {someNumber: {type: 'number', default: 3}}
+            });
+            expect(errors).toHaveLength(0);
+        });
+        test('should return error if default is not a number', () => {
+            const errors = validateState({
+                key: 'state',
+                value: {someNumber: {type: 'number', default: '3'}}
+            });
+            expect(errors).toHaveLength(1);
+            expect(errors[0].message).toBe('state.someNumber.default: number expected');
+        });
+        test('should return error if value is greater than maximum', () => {
+            const errors = validateState({
+                key: 'state',
+                value: {someNumber: {type: 'number', default: 3, maximum: 2}}
+            });
+            expect(errors).toHaveLength(1);
+            expect(errors[0].message).toBe('state.someNumber.default: must be less than or equal to 2');
+        });
+        test('should return error if value is less than minimum', () => {
+            const errors = validateState({
+                key: 'state',
+                value: {someNumber: {type: 'number', default: 1, minimum: 2}}
+            });
+            expect(errors).toHaveLength(1);
+            expect(errors[0].message).toBe('state.someNumber.default: must be greater than or equal to 2');
+        });
+        test('should return error if minimum is not a number', () => {
+            const errors = validateState({
+                key: 'state',
+                value: {someNumber: {type: 'number', default: 1, minimum: '2'}}
+            });
+            expect(errors).toHaveLength(1);
+            expect(errors[0].message).toBe('state.someNumber.minimum: must be a number');
+        });
```

Comment:
Nit-picking: better having a newline after each test, for better readability.

Reply:
Fixed